src/creature.cpp

```diff
@@ -102,6 +102,12 @@ void Creature::reset()
     reset_bonuses();
     reset_stats();
 }
+
+void Creature::bleed() const
```

Comment:
It's used in three places and looks rather nice there. Why get rid of it?
>  already managed to gain a potential bug: bloodType can return fd_null, which add_splatter will then use to create a field of nothing.

Then the bug is in the `add_field()`. It should have a guard clause since fields of nothing are meaningless anyway.